### PR TITLE
AO3-4921 Expand challenge assignment spec test

### PR DESCRIPTION
**Pull request by cesy**
https://otwarchive.atlassian.net/browse/AO3-4921

Increases coverage a little, though there's still some way to go.

### CODE REVIEW

File: spec/controllers/challenge_assignments_controller_spec.rb [hunk 1]
```diff
@@ -15,12 +15,62 @@
   end
 
   context "when logged in" do
+    let(:collection2) { create(:collection, challenge: GiftExchange.new) }
```
Comment (redsummernight) [inline spec/controllers/challenge_assignments_controller_spec.rb]: You should use the factory for gift_exchange added in #2756.

**Comment by cesy** [comment]
Fixed, thanks for the suggestion.

File: spec/controllers/challenge_assignments_controller_spec.rb [hunk 2]
```diff
@@ -15,12 +15,63 @@
   end
 
   context "when logged in" do
+    let(:gift_exchange) { create(:gift_exchange) }
+    let(:collection2) { create(:collection, challenge: gift_exchange, challenge_type: "GiftExchange") }
```
Comment (redsummernight) [inline spec/controllers/challenge_assignments_controller_spec.rb]: I think you can move some of the lets inside their specific describes, and avoid using number suffixes.

**Comment by redsummernight** [comment]
LGTM. :+1:

File: spec/controllers/challenge_assignments_controller_spec.rb (hunk 2 above)
Comment (cesy) [inline spec/controllers/challenge_assignments_controller_spec.rb]: I'll do this in the expansion I'm working on, since I'm also adding some more checks.

**Comment by scottsds** [comment]
👍

**Pull request merged**